src/common/components/nav/nav.component.js

```diff
@@ -127,7 +129,15 @@ class NavController{
   }
 
   signOut() {
-    this.sessionService.downgradeToGuest().subscribe(angular.noop);
+    let modal = this.$uibModal.open({
+      templateUrl: signOutTemplate.name,
+      backdrop: 'static',
+      keyboard: false,
+      size: 'sm'
+    });
+    this.sessionService.downgradeToGuest().subscribe(() => {
+      modal.close();
+    }, angular.noop);
```

Comment:
Idk what could cause an error but should we close the modal on error too? I guess leaving it there would force the user to refresh the page as it stops them from doing anything... But then they still might not be signed out.